sklearn/ensemble/_hist_gradient_boosting/splitting.pyx

```diff
@@ -9,12 +9,12 @@
 
 cimport cython
 from cython.parallel import prange
-cimport numpy as cnp
 import numpy as np
 from libc.math cimport INFINITY, ceil
 from libc.stdlib cimport malloc, free, qsort
 from libc.string cimport memcpy
 
+from sklearn.utils._typedefs cimport uint8_t
```

Comment:
I find it weird to mix relative and absolute sklearn imports in the same Cython file.

I think our conventions are to always use relative imports in non-test files.

Reply:
According to [PEP 8](https://peps.python.org/pep-0008/#imports), absolute imports are recommended.
I tried to use the same kind as already present. So I'll fix this one (soon).

Reply:
> According to PEP 8, absolute imports are recommended (but this may be outdated, on the other side there is a ruff rule for it).

But we have already a convention in place for scikit-learn: "use relative imports in non-test files under the `sklearn.` namespace". This convention has been applied since 10 years.

Maybe it has little value, but I also see little value in changing it now and not having a convention at all has even lower value.

Reply:
fixed